api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1859,6 +1859,97 @@ type AzurePlatformSpec struct {
 	// +immutable
 	// +required
 	SecurityGroupID string `json:"securityGroupID,omitempty"`
+
+	// ManagedIdentities contains the client IDs related to the managed identities needed for HCP control plane
+	// and data plane components that authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
+	// +required
```

Comment:
You only need one of these, typically we've gone for the kubebuilder version but the very latest release now accepts both. You should try to be consistent across your codebase and choose one